src/base/LruMap.h

```diff
@@ -148,39 +169,47 @@ LruMap<EntryValue, EntryCost>::add(const char *key, EntryValue *t)
     if (memLimit() == 0)
         return false;
 
-    index.push_front(new Entry(key, t));
+    auto e = new Entry(key, t, ttl);
+    memUsed_ += e->memCost() + e->key.length() + sizeof(Entry) + sizeof(MapPair);
```

Comment:
> Alex: The map does not store MapPairs
> Amos: The storage map stores a MapPair per Entry

I was probably wrong, sorry: AFAICT, the map does store each key-value pair as a single std::pair object, not as separate items.

> [MapPair] consumes memory in almost as significant amounts as the Entry does due to the way key strings may be longer than the Entry data string coming out of helpers

That does not compute for me because sizeof(MapPair) is a small number (~40) that does not include (or depend on) the size of the LruMap key (i.e., bytes allocated and stored by std::string). I certainly agree that a key can consume much more RAM than EntryValue, but we already account for that in `+ e->key.length()`, don't we?

_Terminology_: Here, "payload" is, roughly speaking, the information that the LruMap class user wants to store/cache. Everything else is "overhead" required to store that information. Most of that overhead, including its most significant portions (e.g., overallocated memory to reduce the number of hash collisions) are invisible to us -- they are hidden inside STL objects.

I think we should calculate just the payload size. That size is `e->memCost() + e->key.length()`. There are many other memory overheads but they are virtually impossible to compute correctly, and they should be marginal compared to the payload size anyway. The code that attempts to include some overheads but does not include many other overheads looks inconsistent or confused. If we could measure everything, it could make sense to include all storage overheads as well, but since we cannot, we should stick to measuring payload IMO.

If the `sizeof(Entry) + sizeof(MapPair)` component is meant to represent something other than overheads, please add a comment to explain its role.